### PR TITLE
Add config for save.title and save.description, fix URIHash, add StringHash

### PR DESCRIPTION
**Pull request by Brewer-Slack**
Added save.title and save.description configs. Will work on tags next. 

StringHash is added as well.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -149,6 +149,12 @@
             <version>2.23.4</version>
         </dependency>
 
+        <dependency>
+            <groupId>com.googlecode.json-simple</groupId>
```
Comment (aniehuser) [inline pom.xml]: where is this dependency used? we shouldn't import external dependencies if they aren't used

File: src/main/java/org/yesworkflow/save/HttpSaver.java [hunk 2]
```diff
@@ -1,5 +1,6 @@
 package org.yesworkflow.save;
 
+import org.mockito.internal.runners.StrictRunner;
```
Comment (aniehuser) [inline src/main/java/org/yesworkflow/save/HttpSaver.java]: don't include this in source code, mockito is used for our unit testing

File: pom.xml (hunk 1 above)
Comment (Brewer-Slack) [reply pom.xml]: Oh I believe that was left over from old serialization. I'll remove it.

File: src/main/java/org/yesworkflow/save/StringHash.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package org.yesworkflow.save;
+
+public class StringHash {
+
+    public static String getHash(String hashString) {
```
Comment (aniehuser) [inline src/main/java/org/yesworkflow/save/StringHash.java]: we should put this method in the other class you made

File: src/main/java/org/yesworkflow/save/HttpSaver.java (hunk 2 above)
Comment (Brewer-Slack) [reply src/main/java/org/yesworkflow/save/HttpSaver.java]: Got it

File: src/main/java/org/yesworkflow/save/StringHash.java (hunk 3 above)
Comment (Brewer-Slack) [reply src/main/java/org/yesworkflow/save/StringHash.java]: ok

Comment (aniehuser) [reply src/main/java/org/yesworkflow/save/StringHash.java]: `URIHash`

File: src/main/java/org/yesworkflow/save/URIHash.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+    private MessageDigest md;
+
+    public URIHash(String hashAlgorithm) throws NoSuchAlgorithmException{
+
+        this.md = MessageDigest.getInstance(hashAlgorithm);
+
+    }
+
+    public String getHash(Path filePath) throws IOException {
```
Comment (aniehuser) [inline src/main/java/org/yesworkflow/save/URIHash.java]: Each of these methods uses redundant code:
```
        byte[] buffer = new byte[1024];
        int inputRead;
        while ((inputRead = inputStream.read(buffer)) != -1) {
            md.update(buffer, 0, inputRead);
        }

        StringBuilder hexString = new StringBuilder();
        for(byte b: md.digest()) {
            hexString.append(String.format("%02x", b));
        }

        return hexString.toString();
```

We should consolidate this into a single private method called something like `hashInputStream`

File: src/main/java/org/yesworkflow/save/URIHash.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import java.net.URI;
+import java.nio.file.Path;
+import java.security.MessageDigest;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.NoSuchAlgorithmException;
+
+
+
+public class URIHash {
```
Comment (aniehuser) [inline src/main/java/org/yesworkflow/save/URIHash.java]: If we add the string hash method to this class, we should rename the class

Comment (Brewer-Slack) [reply src/main/java/org/yesworkflow/save/URIHash.java]: Would renaming to Hash be too un-descriptive?

File: src/main/java/org/yesworkflow/save/URIHash.java (hunk 4 above)
Comment (Brewer-Slack) [reply src/main/java/org/yesworkflow/save/URIHash.java]: We originally only had one method. It was suggested earlier that we handle other types, which is why there are three methods.

**Comment by Brewer-Slack** [comment]
URIHash.java has been refactored to Hash.java. StringHash.java's method moved to Hash.java

File: src/main/java/org/yesworkflow/save/URIHash.java (hunk 4 above)
Comment (aniehuser) [reply src/main/java/org/yesworkflow/save/URIHash.java]: you can make a `private` method do reduce the redundancy that i pointed out above

File: src/main/java/org/yesworkflow/save/URIHash.java (hunk 5 above)
Comment (aniehuser) [reply src/main/java/org/yesworkflow/save/URIHash.java]: thats fine in my opinion

**Comment by Brewer-Slack** [comment]
Should all be finished up. Tests are added and verified for all methods.

**Pull request merged**